src/pipdeptree/_discovery.py

```diff
@@ -1,35 +1,29 @@
 from __future__ import annotations
 
+import site
+import sys
+from importlib.metadata import distributions
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from pip._vendor.pkg_resources import DistInfoDistribution
+    from importlib.metadata import Distribution
 
 
 def get_installed_distributions(
     local_only: bool = False,  # noqa: FBT001, FBT002
     user_only: bool = False,  # noqa: FBT001, FBT002
-) -> list[DistInfoDistribution]:
-    try:
-        from pip._internal.metadata import pkg_resources  # noqa: PLC0415, PLC2701
-    except ImportError:
-        # For backward compatibility with python ver. 2.7 and pip
-        # version 20.3.4 (the latest pip version that works with python
-        # version 2.7)
-        from pip._internal.utils import misc  # noqa: PLC0415, PLC2701 # pragma: no cover
-
-        return misc.get_installed_distributions(  # type: ignore[no-any-return,attr-defined]
-            local_only=local_only,
-            user_only=user_only,
-        )
-
-    else:
-        dists = pkg_resources.Environment.from_paths(None).iter_installed_distributions(
-            local_only=local_only,
-            skip=(),
-            user_only=user_only,
-        )
-        return [d._dist for d in dists]  # type: ignore[attr-defined] # noqa: SLF001
+) -> list[Distribution]:
+    if user_only:
+        return list(distributions(path=[site.getusersitepackages()]))
+
+    # NOTE: See https://docs.python.org/3/library/venv.html#how-venvs-work for more details.
+    in_venv = sys.prefix != sys.base_prefix
+
+    if local_only and in_venv:
+        venv_site_packages = list(site.getsitepackages(sys.prefix) or [])
```

Comment:
You are passing a `str` as an `Iterable` when it should be a `list[str]`:
```suggestion
        venv_site_packages = site.getsitepackages([sys.prefix])
```

Weirdly enough the docs don't show a `prefix` parameter for `getsitepackages()` but looking at the source, it seems to have been introduced between Python v3.2 and v3.3 (so a long time ago). I think it should be fine to keep it.

Reply:
I made this change after checking the [source code](https://github.com/python/cpython/blob/bfc57d43d8766120ba0c8f3f6d7b2ac681a81d8a/Lib/site.py#L362). But you are right, `getsitepackages` should get a `list` instead of a single `str`.
But it looks running well? That's truely weird.

Reply:
Done

Reply:
> But you are right, `getsitepackages` should get a `list` instead of a single `str`. But it looks running well? That's truely weird.

If you talking about `test_local_only()`, that's because when you monkeypatch `sys.prefix`, you assign a list of strings which is correct.